plugin/trino-base-jdbc/src/test/java/io/trino/plugin/jdbc/BaseJdbcConnectorTest.java

```diff
@@ -1548,4 +1549,75 @@ public static Object[][] batchSizeAndTotalNumberOfRowsToInsertDataProvider()
                 {10, 52}, // number of rows > n * batch size
         };
     }
+
+    // test polymorphic table function
+
+    @Test
+    public void testNativeQuerySimple()
+    {
+        assertThat(query("SELECT * FROM TABLE(" + getSession().getCatalog().orElseThrow() + ".system.query(query => 'SELECT 1'))"))
```

Comment:
I'd capture `getSession().getCatalog().orElseThrow()` as a variable
however, you don't need this at all:
```sql
... FROM TABLE(systen.query(....))
```